### PR TITLE
[#1] Implement shared/Button component

### PR DESCRIPTION
**Pull request by TomomiKimura**

### CODE REVIEW

File: apps/docs/pages/index.tsx [hunk 1]
```diff
@@ -1,12 +1,36 @@
...
+            </Button>
+
+            <Button
+                type="destructive"
+                onClick={() => console.log("I also got clicked!")}
+            >
+                investigate service agreement options
+            </Button>
+        </div>
+    );
```
[the remark this answers is not in the document]
Comment (TomomiKimura) [reply apps/docs/pages/index.tsx:35]: Thank you for your comment. I think I understand better now about Storybook compare to the time I made this PR. I made the changes!

File: packages/ui/.storybook/preview.js [hunk 2]
```diff
@@ -1,12 +1,12 @@
 // Its saying that to storybook to make css module available from anywhere.
-import "ui/shared/variables.css";
+import "../shared/variables.css";
```
[the remark this answers is not in the document]
Comment (TomomiKimura) [reply packages/ui/.storybook/preview.js:2]: Thank you! I made the change!

File: packages/ui/shared/variables.css [hunk 3]
```diff
@@ -29,4 +36,9 @@
     --font-h4: 24px;
     --font-h5: 18px;
     --font-h6: 14px;
-}
\ No newline at end of file
+    --font-h7: 12px;
+
+    /* === Font family === */
+    /* It seems like this font is not free. */
+    --font-family: GreyCliffCF-Heavy;
```
[the remark this answers is not in the document]
Comment (TomomiKimura) [reply packages/ui/shared/variables.css:43]: Yes you are right. I decided not to use it, so I deleted it!

File: packages/ui/shared/Button/Button.tsx [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+import styles from "./Button.module.css";
+
+export type ButtonProps = {
+    type: "primary" | "secondary" | "subtle" | "destructive";
+    children: string;
+    onClick?: () => void;
+    disabled?: boolean;
+};
+
+export const Button: React.FC<ButtonProps> = ({
```
[the remark this answers is not in the document]
Comment (TomomiKimura) [reply packages/ui/shared/Button/Button.tsx:15]: After adding prettier-ignore yesterday, it got fixed!

File: packages/ui/.storybook/preview.js [hunk 5]
```diff
@@ -1,12 +1,12 @@
...
 
 export const parameters = {
-  actions: { argTypesRegex: "^on[A-Z].*" },
-  controls: {
-    matchers: {
-      color: /(background|color)$/i,
-      date: /Date$/,
+    actions: { argTypesRegex: "^on[A-Z].*" },
```
[the remark this answers is not in the document]
Comment (TomomiKimura) [reply packages/ui/.storybook/preview.js:5]: It's fixed after adding prettier-ignore!

File: packages/ui/shared/Button/Button.tsx [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+// You need to import file from a file, in this case, Button.module.css. Since we don't know the final name of button, you cannot use just className="button". it should be dynamic {styles.button}
+// So it's creating a variable and passing as a variable {styles.button}
+import styles from "./Button.module.css";
+
+export type ButtonProps = {
+    type: "primary" | "secondary" | "subtle" | "destructive";
+    children: string;
+    onClick?: () => void;
+    disabled?: boolean;
+};
```
[the remark this answers is not in the document]
Comment (TomomiKimura) [reply packages/ui/shared/Button/Button.tsx:13]: I appreciate your time! Let me know when you are available!

File: packages/ui/shared/variables.css [hunk 7]
```diff
@@ -2,17 +2,24 @@
...
     --colors-secondary-text: #ffffff;
+    --colors-secondary-hover: #ba3d07;
+    --colors-subtle: #f0efee;
+    --colors-subtle-text: #444444;
+    --colors-subtle-hover: #d9d3d3;
     --colors-error: #ff0000;
     --colors-error-text: #ffffff;
+    --colors-error-hover: #b40708;
+    --colors-disabled: #e5e6e7;
+    --colors-disabled-text: #9ea0a1;
```
[the remark this answers is not in the document]
Comment (TomomiKimura) [reply packages/ui/shared/variables.css:16]: I agree! Thank you for sharing your thoughts for improvement!

**Review by MiguelAraCo — APPROVED** [review]
LGTM! Thank you for adding the TODOs. Let's address that as a separate effort.

**Pull request merged**